### PR TITLE
keyspace: add keyspace util func and ut

### PR DESCRIPTION
**Pull request by ystaticy**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/41354

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: keyspace/keyspace_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+package keyspace
+
+import (
+	"testing"
+
+	"github.com/pingcap/tidb/config"
+	"github.com/stretchr/testify/suite"
+)
+
+type keyspaceSuite struct {
```
Comment (hawkingrei) [inline keyspace/keyspace_test.go:24]: Please run ```make bazel_prepare``` and set ```flaky = True``` at the ```go_test``` in the bazel config

**Review by tangenta — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**